src/lib/components/versions/VersionDependenciesGrid.svelte

```diff
@@ -1,35 +1,62 @@
 <script lang="ts">
-  import { base } from '$app/paths';
+  import DependencyRow from './DependencyRow.svelte';
+
   import type { VersionDependency } from '$lib/generated';
 
-  export let dependencies!: Pick<VersionDependency, 'mod_id' | 'condition'>[];
+  export let dependencies!: Pick<VersionDependency, 'mod_id' | 'optional' | 'condition'>[];
+
+  const requiredDependencies: Pick<VersionDependency, 'mod_id' | 'optional' | 'condition'>[] = [];
+  const optionalDependencies: Pick<VersionDependency, 'mod_id' | 'optional' | 'condition'>[] = [];
+  dependencies.forEach((dependency) => {
+    if (dependency.optional) {
+      optionalDependencies.push(dependency);
+    } else {
+      requiredDependencies.push(dependency);
+    }
+  });
```

Comment:
These need to be reactive, as the component can be reused when navigating from one mod to another.
```suggestion
  $: requiredDependencies = dependencies.filter((d) => !d.optional);
  $: optionalDependencies = dependencies.filter((d) => d.optional);
```